src/ArgumentResolver/CompatibleValueResolverInterface.php

```diff
@@ -17,14 +17,14 @@
 use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
 
 // TODO: Remove this interface when dropping support of Symfony < 6.2 and replace its usage with ValueResolverInterface
-if (interface_exists(ValueResolverInterface::class)) {
+if (interface_exists(ArgumentValueResolverInterface::class)) {
     /** @internal */
-    interface CompatibleValueResolverInterface extends ValueResolverInterface
+    interface CompatibleValueResolverInterface extends ArgumentValueResolverInterface
```

Comment:
@VincentLanglet isn't it the other way around?

this code gives me deprecations right now

Reply:
Can you try
```
if (interface_exists(ArgumentValueResolverInterface::class) && !interface_exists(ValueResolverInterface::class)) {
    /** @internal */
    interface CompatibleValueResolverInterface extends ArgumentValueResolverInterface
    {
    }
} else (interface_exists(ValueResolverInterface::class)) {
    /** @internal */
    interface CompatibleValueResolverInterface extends ValueResolverInterface
    {
    }
}
```
?

If it works for you PR is welcome

Reply:
The Question is more about what to do between the versions:

* 3.1.0 - 6.2.0 there was only ArgumentValueResolverInterface
* 6.2.0 - 7.0.0 there are both ArgumentValueResolverInterface and ValueResolverInterface, the first one is deprecated.
* 7.0.0 - there is only ValueResolverInterface

what would be the right choice for 6.2.0 - 7.0.0 ? Which Interface should we extend?



as it was before, using ValueResolverInterface if able was better?

Reply:
Feel free to open a pr with what you have in mind and try having a green build

Reply:
this should work:
https://github.com/sonata-project/SonataAdminBundle/pull/8139